### PR TITLE
✨ feat(rust/hashing): add array_hash_map

### PR DESCRIPTION
**Pull request by xBLACKICEx**
If this PR is related to coding or code translation, please read the [contribution guideline](https://github.com/krahets/hello-algo/issues/15#issue-1464069591), fill out the checklist, and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).
image

### CODE REVIEW

**Comment by krahets** [comment]
To be reviewed by @sjinzh

File: codes/rust/chapter_hashing/array_hash_map.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    /* 打印哈希表 */
+    pub fn print(&self) {
+        for entry in self.entry_set() {
+            println!("{} -> {}", entry.key, entry.val);
+        }
+    }
+}
+
+fn main() {
+    // /* 初始化哈希表 */
```
Comment (sjinzh) [inline codes/rust/chapter_hashing/array_hash_map.rs:73]: /* 初始化哈希表 */

File: codes/rust/chapter_hashing/array_hash_map.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    let mut map = ArrayHashMap::new();
+    //
+    // /*添加操作 */
+    // // 在哈希表中添加键值对 （key, value）
```
Comment (sjinzh) [inline codes/rust/chapter_hashing/array_hash_map.rs:77]: /*添加操作 */
// 在哈希表中添加键值对 （key, value）

**Review by sjinzh — CHANGES_REQUESTED** [review]

**Review by sjinzh — APPROVED** [review]
@krahets  It's OK.

**Review by krahets — APPROVED** [review]

**Pull request merged**